File: backend/app/schemas/grocery.py
import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class GroceryItemCreate(BaseModel):
    item: str
    quantity: str | None = None


class GroceryItemUpdate(BaseModel):
    item: str | None = None
    quantity: str | None = None
    checked: bool | None = None


class GroceryItemResponse(BaseModel):
    id: UUID
    item: str
    quantity: str | None
    checked: bool
    created_at: datetime.datetime
    updated_at: datetime.datetime

    model_config = ConfigDict(from_attributes=True)


class ClearCheckedResponse(BaseModel):
    deleted_count: int

File: frontend/types/grocery.ts
export interface GroceryItemResponse {
  id: string
  item: string
  quantity: string | null
  checked: boolean
  created_at: string
  updated_at: string
}

export interface GroceryItemCreateInput {
  item: string
  quantity?: string
}

export interface GroceryItemUpdateInput {
  item?: string
  quantity?: string
  checked?: boolean
}

export interface ClearCheckedResponse {
  deleted_count: number
}
